app/ante/reject_msgs.go

```diff
@@ -9,28 +9,43 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/authz"
 	govtypesv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
 	"github.com/cosmos/cosmos-sdk/x/group"
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
 	evmtypes "github.com/evmos/ethermint/x/evm/types"
 )
 
 // RejectMessagesDecorator prevents invalid msg types from being executed
 type RejectMessagesDecorator struct {
-	disabledMsgTypeURLs map[string]struct{}
+	// message is rejected if any Predicate returns true
+	predicates []Predicate
 }
 
-var _ sdk.AnteDecorator = RejectMessagesDecorator{}
+// Predicate should return true if message is not allowed
+type Predicate = func(typeURL string, depth int) bool
 
-// NewRejectMessagesDecorator creates a decorator to block provided messages from reaching the mempool
-func NewRejectMessagesDecorator(disabledMsgTypeURLs ...string) RejectMessagesDecorator {
-	disabledMsgsMap := make(map[string]struct{})
-	for _, url := range disabledMsgTypeURLs {
-		disabledMsgsMap[url] = struct{}{}
+func BlockTypeUrls(typeUrls ...string) Predicate {
+	block := make(map[string]struct{})
+	for _, url := range typeUrls {
+		block[url] = struct{}{}
+	}
+	return func(url string, depth int) bool {
```

Comment:
```suggestion
	return func(url string, _ int) bool {
```
depth is omitted here